src/main/java/org/liftoff/saintlouisfarms/controllers/StoreController.java

```diff
@@ -79,54 +79,109 @@ public  String displaySpecificFarmNameWithProduct(Model model,
        HttpSession session = request.getSession(false);
        if(session != null){
 //          Should see if they have an active shopping basket
-           Client client = authenticationController.getClientFromSession(session);
+           Client client= authenticationController.getClientFromSession(session);
            model.addAttribute("loggedIn", client != null);
+           //check if the this is the first time for the client and he doesnt have a cart
+           if(shoppingBasketRepository.findAboutClientCart(client.getId()) == null){
            shoppingBasket = new ShoppingBasket(client, LocalDateTime.now());
+           }
+           else{
+               shoppingBasket=shoppingBasketRepository.findAboutClientCart(client.getId());
+           }
+           ShoppingBasketDTO shoppingBasketDTO = new ShoppingBasketDTO();
+
+           shoppingBasketRepository.save(shoppingBasket);
+           model.addAttribute("fa", farmName);
+           if(info!=null){
```

Comment:
I am not seeing where this is getting passed in.

Reply:
I think search should be implemented in the post handler for this page and the get handler. This is why you are redirected to the home page if you search with something in your basket.